### PR TITLE
Added a proof of concept to examples/kitchen-sink for RUM

### PR DESCRIPTION
**Pull request by adambraimbridge**
image

Todo

- [ ] Run the observers for only a small random selection of users
- [ ] Track Time-to-interactive (TTI) performance events
- [ ] Collect appropriate performance events into an object that can be sent to o-tracking in one go
- [ ] Figure out how to track `longtask` performance events
- [ ] Documentation
- [ ] Testing

### CODE REVIEW

**Comment by adambraimbridge** [comment]
image

File: packages/dotcom-ui-real-user-monitoring/src/browser.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  // Get data from the navigation performance entry
+  const { type, domInteractive, domComplete, responseStart, requestStart } = entries.find(
+    (entry) => entry.entryType === 'navigation'
+  )
+
+  console.log({ type }) // if (type !== 'navigate') return
+
+  const timeToFirstByte = responseStart - requestStart
+  const firstPaint = entries.find((entry) => entry.name === 'first-paint').startTime
```
Comment (adambraimbridge) [inline packages/dotcom-ui-real-user-monitoring/src/browser.ts]: we also want firstContentfulPaint

File: packages/dotcom-ui-real-user-monitoring/src/browser.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    largestContentfulPaintEntry.renderTime || largestContentfulPaintEntry.loadTime
+
+  const performanceContext = {
+    timeToFirstByte,
+    domInteractive,
+    firstPaint,
+    largestContentfulPaint,
+    domComplete,
+    timeToInteractive
+  }
```
Comment (adambraimbridge) [inline packages/dotcom-ui-real-user-monitoring/src/browser.ts]: Round up each metric to the nearest millisecond integer

**Comment by i-like-robots** [comment]
Closing as this was implemented in https://github.com/Financial-Times/n-tracking/pull/27 and https://github.com/Financial-Times/n-tracking/pull/32

**Pull request closed without merging**